### PR TITLE
README: Include build instructions for including sputnikvm

### PR DESCRIPTION
**Pull request by whilei**

### CODE REVIEW

**Comment by sorpaas** [comment]
And in Windows:

```
set CGO_LDFLAGS=-Wl,--allow-multiple-definition %GOPATH%\src\github.com\ethereumproject\sputnikvm-ffi\c\sputnikvm.lib -lws2_32 -luserenv
go build -tags=sputnikvm ./cmd/geth
```

**Comment by sorpaas** [comment]
Hmm, in Windows, the above command would also needs to change. `libsputnikvm.a` file is called `sputnikvm.lib`. And all `/` needs to be changed to `\`.

So I would suggest to use a build system instead of adding instructions in README...

**Comment by whilei** [comment]
Aha, just got your message after the push.

File: README.md [hunk 1]
```diff
@@ -65,6 +65,41 @@ $ cd go-ethereum
...
+```
+In macOS:
+```
+cd $GOPATH/src/github.com/ethereumproject/go-ethereum/cmd/geth
+CGO_LDFLAGS="$GOPATH/src/github.com/ethereumproject/sputnikvm-ffi/c/libsputnikvm.a -ldl -lresolv" go build -tags=sputnikvm .
+```
+
+> And for __Windows__:
+```
+cd $GOPATH\src\github.com/ethereumproject
```
Comment (sorpaas) [inline README.md]: It's `%GOPATH%`.

Also `s////\/`

File: README.md [hunk 2]
```diff
@@ -65,6 +65,41 @@ $ cd go-ethereum
...
+git clone https://github.com/ethereumproject/sputnikvm-ffi
+cd sputnikvm-ffi\c\ffi
+cargo build --release
+cp $GOPATH\src\github.com\ethereumproject\sputnikvm-ffi\c\ffi\target\release\sputnikvm.lib $GOPATH\src\github.com\ethereumproject\sputnikvm-ffi\c\sputnikvm.lib
```
Comment (sorpaas) [inline README.md]: Also `%GOPATH`.

Comment (sorpaas) [reply README.md]: Windows doesn't seem to have `cp`. Should be called `copy`.

**Comment by whilei** [comment]
BTW I agree that an optional build script (mage, make, whatever) would be preferable. We can undo this addition when that materializes, but think we need available instructions for building for the meantime.

File: README.md [hunk 3]
```diff
@@ -65,6 +65,44 @@ $ cd go-ethereum
...
+> For __Non-Windows__:
+
+```
+cd $GOPATH/src/github.com/ethereumproject
+git clone https://github.com/ethereumproject/sputnikvm-ffi
+cd sputnikvm-ffi/c/ffi
+cargo build --release
+cp $GOPATH/src/github.com/ethereumproject/sputnikvm-ffi/c/ffi/target/release/libsputnikvm_ffi.a $GOPATH/src/github.com/ethereumproject/sputnikvm-ffi/c/libsputnikvm.a
+```
+And then build geth with CGO_LDFLAGS:
```
Comment (sorpaas) [inline README.md]: Recommend to add "In Linux, ".

File: README.md [hunk 4]
```diff
@@ -65,6 +65,44 @@ $ cd go-ethereum
...
+```
+cd $GOPATH/src/github.com/ethereumproject/go-ethereum/cmd/geth
+CGO_LDFLAGS="$GOPATH/src/github.com/ethereumproject/sputnikvm-ffi/c/libsputnikvm.a -ldl" go build -tags=sputnikvm .
```
Comment (sorpaas) [inline README.md]: Would suggest to be consistent, either for both win and non-win, all use `go build .` or all use `go build ./cmd/geth`.

**Comment by whilei** [comment]
@sorpass I think both of your changes suggest above have been resolved, correct me if I'm wrong.

**Comment by whilei** [comment]
Build cancelled because we'll need to merge or rebase from master after #413

**Pull request merged**